.github/workflows/post.yml

```diff
@@ -50,18 +50,25 @@ jobs:
 
     - name: Build Docker image
       run: |
+        IMAGE_VERSION="latest"
         if [[ "$GITHUB_REF" == refs/tags/* ]]; then
-        IMAGE_VERSION=${GITHUB_REF#refs/tags/}
-        docker build -t ${{ secrets.DOCKER_HUB_USERNAME }}/post-service:${IMAGE_VERSION} -f ./backend/src/PostService/Dockerfile ./backend
-        else
-          echo "Skipping Docker image build because this is not a tag push."
+          IMAGE_VERSION=${GITHUB_REF#refs/tags/}
         fi
+        docker build -t ${{ secrets.DOCKER_HUB_USERNAME }}/post-service:${IMAGE_VERSION} -f ./backend/src/PostService/Dockerfile ./backend/src
 
     - name: Push Docker image to Docker Hub
+      if: startsWith(github.ref, 'refs/tags/')
       run: |
-        if [[ "$GITHUB_REF" == refs/tags/* ]]; then
-          IMAGE_VERSION=${GITHUB_REF#refs/tags/}
-          docker push ${{ secrets.DOCKER_HUB_USERNAME }}/post-service:${IMAGE_VERSION}
-        else
-          echo "Skipping Docker image push because this is not a tag push."
-        fi
\ No newline at end of file
+        IMAGE_VERSION=${GITHUB_REF#refs/tags/}
+        docker push ${{ secrets.DOCKER_HUB_USERNAME }}/post-service:${IMAGE_VERSION}
+        
+  docker-compose:
+    needs: docker
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        
+      - name: Build Docker Compose
+        run: docker compose build --no-cache
```

Comment:
Add some docker compose up and curl step to health check that docker compose actually works